pkg/controller/masteruserrecord/sync.go

```diff
@@ -45,20 +45,39 @@ type Synchronizer struct {
 }
 
 func (s *Synchronizer) synchronizeSpec() error {
-	if !reflect.DeepEqual(s.memberUserAcc.Spec, s.recordSpecUserAcc.Spec) {
-		// when UserAccount spec in record is updated - is not same as in member
-		s.memberUserAcc.Spec = s.recordSpecUserAcc.Spec
+
+	if !s.isSynchronized() {
 		if err := updateStatusConditions(s.hostClient, s.record, toBeNotReady(updatingReason, "")); err != nil {
 			return err
 		}
+		s.memberUserAcc.Spec.UserAccountSpecBase = s.recordSpecUserAcc.Spec.UserAccountSpecBase
+		s.memberUserAcc.Spec.Disabled = s.record.Spec.Disabled
+		s.memberUserAcc.Spec.UserID = s.record.Spec.UserID
 		err := s.memberCluster.Client.Update(context.TODO(), s.memberUserAcc)
 		if err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
 
+func (s *Synchronizer) isSynchronized() bool {
+	if !reflect.DeepEqual(s.memberUserAcc.Spec.UserAccountSpecBase, s.recordSpecUserAcc.Spec.UserAccountSpecBase) {
```

Comment:
I believe you could simply do `return reflect.DeepEqual(s.memberUserAcc.Spec, s.recordSpecUserAcc.Spec)`

Reply:
@xcoulon no. There are two different structs with different fields. It wouldn't work. See https://github.com/codeready-toolchain/host-operator/pull/146/files#r379108646